### PR TITLE
Crypto: add feature to disable installation of built-in provider

### PR DESCRIPTION
**Pull request by stormshield-gt**
Fix https://github.com/rustls/rustls/issues/1920 , especially the problematic described in https://github.com/rustls/rustls/issues/1920#issuecomment-2255255668 .

### CODE REVIEW

**Comment by ctz** [comment]
Addressing the coverage issue in #2083 (preexisting issue on main).

**Comment by ctz** [comment]
This looks reasonable to me.

It would be good to add some documentation for this to https://rustls.github.io/rustls/prerelease/#crate-features, and perhaps a short mention in https://rustls.github.io/rustls/prerelease/crypto/struct.CryptoProvider.html#using-the-per-process-default-cryptoprovider .

**Comment by stormshield-gt** [comment]
I added some comments where indicated, I'm not a native English speaker, so the wording might not be perfect.

File: rustls/Cargo.toml [hunk 1]
```diff
@@ -40,6 +40,7 @@ aws_lc_rs = ["dep:aws-lc-rs", "webpki/aws_lc_rs"]
 aws-lc-rs = ["aws_lc_rs"] # Alias because Cargo features commonly use `-`
 brotli = ["dep:brotli", "dep:brotli-decompressor", "std"]
 ring = ["dep:ring", "webpki/ring"]
+custom_provider = []
```
Comment (djc) [inline rustls/Cargo.toml:43]: I think we wanted to standardize on dashes over underscores in feature names now?

**Review by djc — APPROVED** [review]

File: rustls/Cargo.toml (hunk 1 above)
Comment (stormshield-gt) [reply rustls/Cargo.toml:43]: https://github.com/rustls/rustls/issues/1961 seems to go in that direction but don't mention the `read_buf` feature so I was confused

Comment (djc) [reply rustls/Cargo.toml:43]: Probably still use a dash here.

Comment (stormshield-gt) [reply rustls/Cargo.toml]: I've changed it to a dash in the latest push ;)

**Comment by djc** [comment]
@ctz do you want to take another look at this?

**Review by ctz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**